siac/parse_test.go

```diff
@@ -0,0 +1,141 @@
+package main
+
+import (
+	"testing"
+)
+
+func TestNoSuffix(t *testing.T) {
+	a := "123"
+	b, err := parseSize(a)
+	if a != b || err != nil {
+		t.Error("ParseSize 123 failed")
+	}
+}
+
+func TestTwoCharSuffix(t *testing.T) {
+	a := "123TB"
+	b, err := parseSize(a)
+	if b != "123000000000000" || err != nil {
+		t.Error("ParseSize 123TB failed")
+	}
+}
+
+func TestThreeCharSuffix(t *testing.T) {
+	a := "123GiB"
+	b, err := parseSize(a)
+	if b != "132070244352" || err != nil {
+		t.Error("ParseSize 123GiB failed")
+	}
+}
+
+func TestBadSuffix(t *testing.T) {
+	a := "123BiB"
+	b, err := parseSize(a)
+	if b != "" || err == nil {
+		t.Error("ParseSize 123BiB failed")
+	}
+}
+
+func TestZeroLen(t *testing.T) {
+	a := ""
+	b, err := parseSize(a)
+	if b != "" || err != nil {
+		t.Error("ParseSize emptystring failed")
+	}
+}
+
+func TestShortSuffix(t *testing.T) {
+	a := "123G"
+	b, err := parseSize(a)
+	if b != "" || err == nil {
+		t.Error("ParseSize 123G failed")
+	}
+}
+
+func TestOnlySuffix(t *testing.T) {
+	a := "GB"
+	b, err := parseSize(a)
+	if b != "" || err == nil {
+		t.Error("ParseSize GB failed")
+	}
+}
+
+func TestNumInSuffix(t *testing.T) {
+	a := "123B99"
+	b, err := parseSize(a)
+	if b != "" || err != nil {
+		t.Error("ParseSize 123B99 failed")
+	}
+}
+
+func TestTB(t *testing.T) {
+	a := "1TB"
+	b, err := parseSize(a)
+	if b != "1000000000000" && err == nil {
+		t.Error("ParseSize 1TB failed")
+	}
+}
+
+func TestGB(t *testing.T) {
+	a := "1GB"
+	b, err := parseSize(a)
+	if b != "1000000000" && err == nil {
+		t.Error("ParseSize 1GB failed")
+	}
+}
+
+func TestMB(t *testing.T) {
+	a := "1MB"
+	b, err := parseSize(a)
+	if b != "1000000" && err == nil {
+		t.Error("ParseSize 1MB failed")
+	}
+}
+
+func TestKB(t *testing.T) {
+	a := "1KB"
+	b, err := parseSize(a)
+	if b != "1000" && err == nil {
+		t.Error("ParseSize 1KB failed")
+	}
+}
+
+func TestTiB(t *testing.T) {
+	a := "1TiB"
+	b, err := parseSize(a)
+	if b != "1099511627776" && err == nil {
+		t.Error("ParseSize 1TiB failed")
+	}
+}
+
+func TestGiB(t *testing.T) {
+	a := "1GiB"
+	b, err := parseSize(a)
+	if b != "1073741824" && err == nil {
+		t.Error("ParseSize 1GiB failed")
+	}
+}
+
+func TestMiB(t *testing.T) {
+	a := "1MiB"
+	b, err := parseSize(a)
+	if b != "1048576" && err == nil {
+		t.Error("ParseSize 1MiB failed")
+	}
+}
+
+func TestKiB(t *testing.T) {
+	a := "1KiB"
+	b, err := parseSize(a)
+	if b != "1024" && err == nil {
+		t.Error("ParseSize 1KiB failed")
+	}
+}
+
+func Test12A3456(t *testing.T) {
+	a := "12A3456"
+	b, err := parseSize(a)
+	if b != "" || err != nil {
```

Comment:
Yes, I actually did that on emptystring test and short prefix. Fixed in the next push.